ENIGMAsystem/SHELL/Graphics_Systems/General/GSvertex_impl.h

```diff
@@ -79,20 +90,26 @@ union VertexElement {
 
 struct VertexBuffer {
   vector<VertexElement> vertices;
-  bool frozen, dirty;
+  bool frozen, dynamic, dirty;
   int format;
   std::size_t number;
 
   VertexBuffer(): frozen(false), dirty(false), format(-1), number(0) {}
+  int getNumber() const {
+    return dirty ? vertices.size() : number;

```

Comment:
No, because `vertex_submit` is guaranteed to clear `vertices` in `graphics_prepare_buffer` once it has moved the vertex data to the GPU. The same story is true for `index_submit` etc.